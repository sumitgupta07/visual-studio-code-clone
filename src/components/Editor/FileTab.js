import React from 'react';
import PropTypes from 'prop-types';

const FileTab = ({ onClick, file, activeFile }) => {
  const handleClick = file => {
    onClick(file);
  };

  let className = 'file-list-item';

  if (activeFile.id === file.id) {
    className += ' file-active';
  }
  return (
    <li id={file.id} className={className} onClick={() => handleClick(file)}>
      {file.fileName}
    </li>
  );
};
File.propTypes = {
  activeFile: PropTypes.object.isRequired,
  file: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};
export default FileTab;
