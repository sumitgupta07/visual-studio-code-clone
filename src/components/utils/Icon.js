import React from 'react';
import './Icon.css';

const Icon = ({ label, onSelectIcon, iconClass }) => {
  return (
    <button onClick={() => onSelectIcon(label)}>
      <div className="tooltip">
        <div className="icon">{<i className={iconClass}></i>}</div>
        <div className="tooltiptext">
          <span>{label}</span>
        </div>
      </div>
    </button>
  );
};

export default Icon;
