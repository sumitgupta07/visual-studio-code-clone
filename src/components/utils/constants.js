export const labels = {
  FileExplorer: 'File Explorer',
  Search: 'Search',
  SourceControl: 'Source Control',
  Debug: 'Debug',
  Extensions: 'Extensions'
};

export const classNames = {
  explorerClass: 'far fa-copy fa-3x',
  searchClass: 'fas fa-search fa-rotate fa-3x',
  sourcecontrolClass: 'fas fa-code-branch fa-3x',
  debugClass: 'fas fa-bug fa-3x',
  extensionsClass: 'fas fa-th-large fa-3x'
};

export const folders = [
  {
    id: 1,
    name: 'src',
    files: [
      {
        id: 'a',
        fileName: 'index.js',
        content: 'Placeholder for index.js. It has App.js as child components'
      },
      {
        id: 'b',
        fileName: 'App.js',
        content:
          'Placeholder for App.js. It has Header and Content as child components'
      },
      {
        id: 'c',
        fileName: 'Content.js',
        content:
          'Placeholder for Content.js file. It has Navbar, SelectPane and EditorPane as child components'
      }
    ]
  }
];
