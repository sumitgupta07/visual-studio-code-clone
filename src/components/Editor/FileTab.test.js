import React from 'react';
import { shallow } from 'enzyme';
import FileTab from './FileTab';
import { folders } from '../utils/constants';

let handleClick;
let wrapper;
let mockProps;
beforeEach(() => {
  handleClick = jest.fn();
  mockProps = {
    onClick: handleClick,
    file: {
      id: 'a',
      fileName: 'index.js',
      content: 'Placeholder for index.js. It has App.js as child components'
    },
    activeFile: {
      id: 'b',
      fileName: 'App.js',
      content:
        'Placeholder for App.js. It has Header and Content as child components'
    }
  };
  wrapper = shallow(<FileTab {...mockProps} />);
});

it('should render FileTab component', () => {
  expect(wrapper).toMatchSnapshot();
});
it('should have only 1 li item', () => {
  expect(wrapper.find('li').length).toEqual(1);
});

it('should have render label for file item only and not for activeFile item', () => {
  expect(wrapper.find('#a').length).toEqual(1);
  expect(wrapper.find('#b').length).toEqual(0);
});

it('should call onClick with current "file" as parameter when current "file" is clicked', () => {
  wrapper.find('#a').simulate('click');
  expect(handleClick).toBeCalledWith(mockProps.file);
});
