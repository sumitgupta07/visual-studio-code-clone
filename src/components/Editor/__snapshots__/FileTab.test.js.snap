// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FileTab component 1`] = `
<li
  className="file-list-item"
  id="a"
  onClick={[Function]}
>
  index.js
</li>
`;
